homeassistant/components/mqtt/climate.py

```diff
@@ -223,6 +224,7 @@
             vol.Optional(CONF_TEMP_LOW_STATE_TOPIC): mqtt.valid_subscribe_topic,
             vol.Optional(CONF_TEMP_STATE_TEMPLATE): cv.template,
             vol.Optional(CONF_TEMP_STATE_TOPIC): mqtt.valid_subscribe_topic,
+            vol.Optional(CONF_TEMP_UNIT): cv.string,
```

Comment:
Use `cv.temperature_unit`

Reply:
Thanks for reviewing!  I changed it to `cv.temperature_unit`, but it doesn't work.  The `temperature_unit` validator does not accept the degree symbol required when used as a unit of measurement.
https://github.com/home-assistant/core/blob/dev/homeassistant/helpers/config_validation.py#L468-L475

Reply:
Right, this is the purpose of `cv.temperature_unit`: in the MQTT discovery message or `configuration.yaml`, `C`or `F` is specified, this is then transformed to `°C` or `°F`.

To make it more similar to other integrations which allows configuring temperature unit in the same way, change to use `const.CONF_TEMPERATURE_UNIT` instead of `const.CONF_UNIT_OF_MEASUREMENT`.

Reply:
As that change will basically rewrite this (small) pull request, I think I will just close this.

Reply:
I think the new functionality is good though. I did some adjustment, does it look OK?

Reply:
Thanks.  Let me test it and get back to you.  I find it curious that the `temperature_unit` climate property will work with either returning the degree symbol or without it.  Which is technically correct?

Reply:
The degree symbol is added by `cv.temperature_unit` when the config schema is validated. I think the purpose of `cv.temperature_unit` is to not have to write the degree symbol in config files which doesn't exist on most keyboards.

Reply:
I tested your changes and it works fine.  Thanks for your help.